app/uk/gov/hmrc/cgtpropertydisposals/service/onboarding/AuditService.scala

```diff
@@ -70,38 +73,64 @@ class AuditServiceImpl @Inject()(
     httpStatus: Int,
     body: String,
     path: String
-  )(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Unit = {
-
-    val detail = SubscriptionResponseEvent(
-      httpStatus,
-      body
-    )
-
-    sendEvent(
-      "subscriptionResponse",
-      detail,
-      hc.toAuditTags("subscription-response", path)
-    )
-
-  }
+  )(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Unit =
+    Try {
+      Json.parse(body)
+    } match {
+      case Failure(exception) => {
+        logger.warn(s"{Could not send subscription response audit message: $exception}")
+        sendEvent(
+          "subscriptionResponse",
+          SubscriptionResponseEvent(
+            httpStatus,
+            Json.parse("""{"body" : "failed to get subscription response"}""")
```

Comment:
could we log the body that did come back? something like `"""{"body" : "could not parse body as JSON: $body"}"""` Then   at least we can see what came back. Same thing for the registration response

Reply:
refactored